### DISCUSSION TITLE
Draw or Rectangle for border around string of text?

### DISCUSSION THREAD

**Discussion by fakefarm** [ImageMagick/ImageMagick#2757] [Help]
Hello World!

Would you recommend `draw` or `rectangle` or a different approach to draw a border around an array of text that it prints out on a PDF dimension? I'm stuck and could use some help.

```
['hello', 'world'].each do |word|
  # What goes here?
  # image_magick_border_magick(word)
end
```

Thanks for reading!

![Screen Shot 2020-10-22 at 9 08 25 PM](https://user-images.githubusercontent.com/587705/96945137-ceb69480-14aa-11eb-98b1-099a7a1ed575.png)

**Comment by fmw42** [comment]
One way is the create a text image on a transparent background using label:  Then use -border to draw a border around it. The use convert ... -compose over -composite ... to place that result on your background image.  See https://imagemagick.org/Usage/text/ and https://imagemagick.org/Usage/crop/#border and https://imagemagick.org/Usage/compose/ and https://imagemagick.org/Usage/layers/#convert

**Reply by fakefarm** [reply]
Thank you! I'll read over those and let you know what solves it for me.
